fix(frontend): create evaluation model testset overlapping issue

ashrafchowdury opened the pull request:
### Description

This PR aims to fix the testset overlapping issue in the create new evaluation model, and also added search feature in the component so that finding testsets gets easier.

### Related Issue

Closes [AGE-1055](https://linear.app/agenta/issue/AGE-1055/[bug]-human-create-eval-modals-select-testset-component-overlaping)

mmabrouk reviewed (changes requested):
Thanks @ashrafchowdury 

- I understand this PR adds a search feature right, but I could not understand the testset overlapping issue, the issue does not explain the problem clearly. Can you please explain it and the solution
- The search feature only works in human evaluation but not in automatic evaluation
- The tests are failing

ashrafchowdury commented:
- Fixed the auto-evaluation search 
- Fixed test fails
- This is the testset overlapping issue that is fixed:

![image](https://github.com/user-attachments/assets/1e43d15d-dcd6-4782-a2ea-5b5db9d2dd00)

bekossy reviewed (approved):
LGTM

bekossy reviewed (changes requested):
Thanks for the PR @ashrafchowdury. It looks like the New Evaluation Modal has already been updated, but the Human Evaluation Modal has not. To keep this PR focused, let's limit its scope to updating the Human Evaluation Modal only
